### PR TITLE
Microsoft Teams Chat Handler

### PR DESCRIPTION
**Pull request by MinuraPunchihewa**
## Description

This PR refactors the existing Microsoft Teams integrations and converts it to a Chat handler.

Fixes https://github.com/mindsdb/mindsdb/issues/7529

## Type of change

- [X] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [X] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [X] I have attached a brief loom video or screenshots showcasing the new functionality or change.

https://www.loom.com/share/c18aac43965d454791d5c7533c8246d9?sid=7b059a6d-768a-4142-9aa5-af1cf392b72f

## Checklist:

- [X] My code follows the style guidelines(PEP 8) of MindsDB.
- [X] I have appropriately commented on my code, especially in complex areas.
- [X] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/utilities/api_utilities/microsoft/ms_graph_api_utilities.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+    def _get_group_ids(self):
+        if not self._group_ids:
+            api_url = self._get_api_url("groups")
+            params = {"$select": "id,resourceProvisioningOptions"}
+            groups = self._get_response_value_unsafe(self._make_request(api_url, params=params))
+            self._group_ids = [item["id"] for item in groups if "Team" in item["resourceProvisioningOptions"]]
+        return self._group_ids
+
+    def get_channels(self):
```
Comment (ea-rus) [inline mindsdb/integrations/handlers/utilities/api_utilities/microsoft/ms_graph_api_utilities.py:74]: I think this method and others below should be located in ms_teams handler. They are about the chat api.
We can leave in this class _fetch_data method and other methods which implement protocol. And use it from ms teams handler as base class

Comment (MinuraPunchihewa) [reply mindsdb/integrations/handlers/utilities/api_utilities/microsoft/ms_graph_api_utilities.py:74]: Hey @ea-rus,
That makes sense. I will work on it.

Comment (MinuraPunchihewa) [reply mindsdb/integrations/handlers/utilities/api_utilities/microsoft/ms_graph_api_utilities.py]: Hey @ea-rus,
I have made the requested changes.

File: mindsdb/integrations/handlers/ms_teams_handler/ms_teams_tables.py [hunk 2]
```diff
@@ -33,32 +107,404 @@ def insert(self, query: ast.Insert) -> None:
...
+            api_client.send_channel_message(
+                group_id=message["channelIdentity_teamId"],
+                channel_id=message["channelIdentity_channelId"],
+                message=message["body_content"],
+                subject=message.get("subject")
+            )
+    
+    def get_columns(self) -> list:
+        return [
+            "id",
```
[the remark this answers is not in the document]
Comment (MinuraPunchihewa) [reply mindsdb/integrations/handlers/ms_teams_handler/ms_teams_tables.py]: @ZoranPandovski I have updated this using `BaseSettings` from `pydantic`. Let me know what you think of the implementation.

File: mindsdb/integrations/handlers/ms_teams_handler/ms_graph_api_teams_client.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
+from typing import Optional
+from mindsdb.integrations.handlers.utilities.api_utilities.microsoft.ms_graph_api_utilities import MSGraphAPIBaseClient
+
+
+class MSGraphAPITeamsClient(MSGraphAPIBaseClient):
```
[the remark this answers is not in the document]
Comment (MinuraPunchihewa) [reply mindsdb/integrations/handlers/ms_teams_handler/ms_graph_api_teams_client.py]: @ZoranPandovski I have added docstrings and comments to all of complex aspects of the code. This has been done for the rest of the handler as well.

File: mindsdb/integrations/handlers/ms_teams_handler/ms_teams_handler.py [hunk 4]
```diff
@@ -1,18 +1,29 @@
...
+from mindsdb.integrations.handlers.ms_teams_handler.ms_teams_tables import ChannelsTable, ChannelMessagesTable, ChatsTable, ChatMessagesTable
+from mindsdb.integrations.libs.api_handler import APIChatHandler
 from mindsdb.integrations.libs.response import (
     HandlerStatusResponse as StatusResponse,
 )
 
 from mindsdb.utilities import log
 from mindsdb_sql import parse_sql
 
+DEFAULT_SCOPES = [
```
[the remark this answers is not in the document]
Comment (MinuraPunchihewa) [reply mindsdb/integrations/handlers/ms_teams_handler/ms_teams_handler.py]: @ZoranPandovski Do you mean that these should be mentioned in the `README`? By the way, I have moved these to a configuration class as well.

**Comment by MinuraPunchihewa** [comment]
Hey @ZoranPandovski,
I have now added,

- An updated README
- A few unit tests
- Moved all configurations, including columns and test data to a settings file

I've also fixed some bugs along the way.

**Comment by MinuraPunchihewa** [comment]
Hey @ZoranPandovski,
I have added quite a few mocked test here. Can you please have a look and let me know what you think?
There are about 35. All of them may not be necessary.

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**